procedures/CodeBrowser_utils.ipf

```diff
@@ -166,91 +175,134 @@ End
 Function setGlobalVar(globalVar, numValue)
 	Variable numValue
 	String globalVar
-	DFREF dfr = createDFWithAllParents(pkgFolder)
 
+	DFREF dfr = createDFWithAllParents(pkgFolder)
 	Variable/G dfr:$globalVar
 	NVAR/Z/SDFR=dfr myVar = dfr:$globalVar
-
 	if(!NVAR_Exists(myVar))
-		DebugPrint("global Variable " + globalVar + " failed to create")
+		DebugPrint("failed to create global variable " + globalVar)
 		return 0
-	else
-		myVar = numValue
-		return 1
 	endif
+
+	myVar = numValue
+	if(numtype(numValue) == 0 && getGlobalVar(globalVar) != numValue)
+		// @todo There should be a test case for this
+		Abort "global variable content missmatch."
+	endif
```

Comment:
@ukos-git In which situation would that be triggered?

Reply:
We always want to get back, what we have written.

The numtype was just to quickly skip the NaN comparison.

Reply:
@ukos-git What I meant was: Against which bug/situation are you protecting against? From my understanding of the code the condition in the if clause can never be true. Or is that basically an assertion?

Reply:
Right, it can never be true. Simpler to drop it here.